### DISCUSSION TITLE
New feature: Analyze dependency changes from user created merge/pull requests

### DISCUSSION THREAD

**Discussion by skorhone** [renovatebot/renovate#22185] [Suggest an Idea]
**What would you like Renovate to be able to do?**
I would have renovate bot comment changes to dependencies that user has done in a merge/pull request. Renovate's way of reporting changes to dependencies is extremely user friendly.

**Describe the solution you'd like**
Currently renovate bot does reads dependency information from a single branch and then queries remote repositories for new versions in specified dependencies. Finally it produces a report based on changes it finds between original and updated dependency lists.

For user made pull request analysis, renovate bot could do first step as it does normally. As a second step, it should use exactly the same algorithm that is being used for first step to read dependencies from the target branch. Finally a report should be generated based on comparison results of branches using similar/same algorithm that is currently being used.

Notice: This same algorithm could be leveraged for whitesource bot's analysis. Being able to see what was done and what was the impact would be extremely nice

**Describe alternatives you've considered**
Use some other tool -> haven't just found one. Obviously having two different output formats would make things harder for users.

**Comment by rarkins** [comment]
It might be more challenging than you expect. Renovate would need to kind of "reverse engineer" the PR diff to try to work out what changed. Also right now we only retrieve PRs that we have created so that would need to change too.

**Comment by skorhone** [comment]
Hmm, why would it need to reverse engineer pr? I think both github and gitlab provide apis, where you can get information of both target and source branch. If you just analyze dependency declarations from each branch and compare the two results, you don't need to look what's actually changed in source code

As an algorithm:
1. Fetch source branch
2. List dependencies from source branch
3. Fetch target branch
4. List dependencies from target branch
5. Compare differences between two lists
6. Report
